File: problems/Java/maxConsecutiveOnes3.java
class Solution {
	public int longestOnes(int[] nums, int k) {
		int n = nums.length;
		if (n < 2 && k < 0)
			return n;

		int l = 0, r = 0;

		int maxlen = 0, counter = 0;
		while (r < n) {
			if (nums[r] == 0) {
				counter++;
			}
			while (k < counter) {
				if (nums[l] == 0) {
					counter--;

				}
				l++;
			}
			maxlen = Math.max(maxlen, r - l + 1);
			r++;
		}
		return maxlen;
	}
}

File: problems/Sliding Window/maxConsecutiveOnes3.py
nums = [1, 1, 1, 0, 0, 0, 1, 1, 1, 1, 0]
k = 2

n = len(nums)

l = 0

r = 0
maxSlide = 0
currentSlide = 0
while r < n:
    if(nums[r] == 1):
        currentSlide += 1
        maxSlide = max(maxSlide, currentSlide)
    elif(k > 0):
        currentSlide += 1
        maxSlide = max(maxSlide, currentSlide)
        k -= 1
    else:
        currentSlide = 0
    r += 1
print(maxSlide)
